tests/defer_regress/tests.py

```diff
@@ -104,16 +104,21 @@ def test_basic(self):
             list(SimpleItem.objects.annotate(Count("feature")).only("name")), list
         )
 
-    def test_ticket_16409(self):
-        # Regression for #16409 - make sure defer() and only() work with annotate()
+    def test_annotation_defer_and_only(self):
+        """
+        Make sure defer() and only() work with annotate() (#16409).
+        """
         self.assertIsInstance(
             list(SimpleItem.objects.annotate(Count("feature")).defer("name")), list
         )
         self.assertIsInstance(
             list(SimpleItem.objects.annotate(Count("feature")).only("name")), list
         )
 
-    def test_ticket_23270(self):
+    def test_deferred_model_inheritance(self):
+        """
+        select_related on fields pointing to subclasses works when using defer (#23270).
+        """
```

Comment:
```suggestion
    def test_defer_with_select_related_inherited_model(self):
```
Maybe we move this down next to the other select related tests for defer